### PR TITLE
Dev frontend

### PR DESCRIPTION
**Pull request by mbronstein3**
Merge changes dev-frontend -> master

### CODE REVIEW

File: frontend/src/views/RegistrationBox.jsx [hunk 1]
```diff
@@ -18,108 +17,189 @@ import CustomInput from "components/CustomInput/CustomInput.jsx";
...
-              </GridItem>
-            </GridContainer>
-          </div>
+                      labelText="Confirm Password"
+                      id="regPasswordConfirm"
+                      name="regPasswordConfirm"
+                      onChange={this.handleChange}
+                      formHelperText={
+                        this.state.submitted && !this.state.regPasswordConfirm
+                          ? "Confirm password is required"
```
Comment (Xvx1) [inline frontend/src/views/RegistrationBox.jsx]: Show message for if confirm password doesn't match password?

File: frontend/src/components/API/UserAuth.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        },
+        body: JSON.stringify({ email, password, name })
+      });
+
+      if (res.ok) {
+        const json = await res.json();
+        this.setToken(json.jwt);
+      } else {
+        // User already registered
+        throw new EmailRegisteredError();
```
Comment (dli357) [inline frontend/src/components/API/UserAuth.js]: Add case for 500 internal server error or 400 bad request

File: frontend/src/components/API/UserAuth.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  }
+
+  async renew() {
+    const token = this.getToken();
+    if (!token) return;
+    try {
+      const res = await fetch("/user/renew", {
+        method: "GET",
+        headers: {
+          Authorization: token,
```
Comment (dli357) [inline frontend/src/components/API/UserAuth.js]: Authorization needs to be in "Bearer: <jwt>" format per JWT convention

File: frontend/src/components/API/UserAuth.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+      // something else went wrong
+    }
+  }
+
+  logout() {
+    this.storage.removeItem("userToken");
+  }
+
+  setToken(userToken) {
+    this.storage.setItem("userToken", userToken);
```
Comment (dli357) [inline frontend/src/components/API/UserAuth.js]: Easiest way to adhere to convention is to prepend the "Bearer" here

File: frontend/src/views/RegistrationBox.jsx (hunk 1 above)
Comment (alex-stetson) [reply frontend/src/views/RegistrationBox.jsx]: Fixed in 2007ca7390ac52aa17413dabbc89b5aacbc91e9f

File: frontend/src/components/API/UserAuth.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        body: JSON.stringify({ email, password })
+      });
+      if (res.ok) {
+        const json = await res.json();
+        this.setToken(json.jwt);
+      } else {
+        // API returned unauthorized
+        throw new IncorrectAuthenticationError();
+      }
+    } catch (error) {
```
[the remark this answers is not in the document]
Comment (alex-stetson) [reply frontend/src/components/API/UserAuth.js]: Fixed in 2007ca7390ac52aa17413dabbc89b5aacbc91e9f

File: frontend/src/components/API/UserAuth.js (hunk 3 above)
Comment (alex-stetson) [reply frontend/src/components/API/UserAuth.js]: Fixed in 2007ca7390ac52aa17413dabbc89b5aacbc91e9f

File: frontend/src/components/API/UserAuth.js (hunk 4 above)
Comment (alex-stetson) [reply frontend/src/components/API/UserAuth.js]: Did this in 2007ca7390ac52aa17413dabbc89b5aacbc91e9f

**Pull request merged**